plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/flex/FlexJarMavenProjectStagingDelegate.java

```diff
@@ -49,15 +46,22 @@
 import org.eclipse.m2e.core.MavenPlugin;
 import org.eclipse.m2e.core.project.IMavenProjectFacade;
 import org.eclipse.m2e.core.project.IMavenProjectRegistry;
-import org.eclipse.ui.console.MessageConsoleStream;
 
+/**
+ * Copies a runnable JAR (an App Engine flexible app) to the given staging directory (in addition to
+ * copying {@code app.yaml} handled by the base class) for Maven projects with the App Engine
+ * flexible JAR facet. The JAR is generated by running {@code mvn package} through the Maven launch
+ * configuration mechanism.
+ *
+ * See the Javadoc of {@link FlexStagingDelegate} for additional details.
+ *
+ * @see FlexStagingDelegate
+ * @see StagingDelegate
+ */
 public class FlexJarMavenProjectStagingDelegate extends FlexStagingDelegate {
```

Comment:
This class doesn't seem to be explicitly tied to just `jar` files.  `FlexMavenPackagedProjectStagingDelegate`?

Reply:
That's true. It can actually package WAR files, and the project doesn't actually need the flex facet. I'll make some updates to code and comments.